### PR TITLE
"plot_pareto_front" improvement

### PR DESCRIPTION
**Pull request by 0x41head**
## Motivation

Refer to #2203 

## Description of the changes

- aligning the color with optuna.visualization.plot_optimization_history, such that the red/blue colors represent the best points or any other points in a consistent way

- including a legend, again, similar to optuna.visualization.plot_optimization_history

### CODE REVIEW

**Comment by hvy** [comment]
Thanks, the changes basically LGTM but could you fix the unit tests that are now failing due to this change? https://github.com/optuna/optuna/pull/2355/checks?check_run_id=1909294510

Also, this is a detail but I'm wondering if you know if we can use the exact same blue and red variations.
- https://66150-122299416-gh.circle-artifacts.com/0/docs/build/html/reference/visualization/generated/optuna.visualization.plot_pareto_front.html#optuna.visualization.plot_pareto_front
- https://66150-122299416-gh.circle-artifacts.com/0/docs/build/html/reference/visualization/generated/optuna.visualization.plot_optimization_history.html#optuna.visualization.plot_optimization_history

**Comment by 0x41head** [comment]
Do you mean the red line that comes between the markers in `optuna.visualization.plot_optimization_history` ?

**Comment by hvy** [comment]
Sorry for my poor explanation. I meant that the blue color in the Pareto front plot and the blue color in the optimization history have seemingly different RGB values. Similarly the red color.

**Comment by 0x41head** [comment]
@hvy  Changed the rgb values of the markers to match `optuna.visualization.plot_optimization_history`
Reference picture : 
image

File: tests/visualization_tests/test_pareto_front.py [hunk 1]
```diff
@@ -36,24 +36,25 @@ def test_plot_pareto_front_2d(
         include_dominated_trials=include_dominated_trials,
         axis_order=axis_order,
     )
-    assert len(figure.data) == 1
+    print(figure.data)
```
Comment (hvy) [inline tests/visualization_tests/test_pareto_front.py:39]: Ops, debug print left.
```suggestion
```

**Review by hvy — CHANGES_REQUESTED** [review]

File: tests/visualization_tests/test_pareto_front.py (hunk 1 above)
Comment (0x41head) [reply tests/visualization_tests/test_pareto_front.py:39]: Thanks for pointing it out .

**Review by hvy — APPROVED** [review]
Thanks for your quick action as always, changes LGTM!

**Review by HideakiImamura — APPROVED** [review]
Thanks for the PR. LGTM!

**Pull request merged**